extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+
+Using `[class]="theFood[currentMeal].style"` as an example of **expression** syntax evaluation:
 
-<button on="tap:AMP.setState({currentAnimal: 'cat'})">Set to Cat</button>
+- `[class]` is the property to update
+- `theFood` is the id of the `<amp-state>` component.
+- `currentMeal` is the state name. In the case of `theFood` it will be `cupcakes` or `sushi`.
+- `style` is the **state variable**. It corresponds to the matching JSON key, and sets the bound property to that key's value.
+
+[/filter] <!-- formats="email" -->
+
+### `<amp-state>` specification
+
+[filter formats="websites, stories, ads"]
+An `amp-state` element may contain either a child `<script>` element **OR** a `src` attribute containing a CORS URL to a remote JSON endpoint, but not both.
+
+```html
+<amp-state id="myLocalState">
+  <script type="application/json">
+    {
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+
+<amp-state id="myRemoteState" src="https://data.com/articles.json"> </amp-state>
 ```
 
 [/filter] <!-- formats="websites, stories, ads" -->
 
 [filter formats="email"]
+An `amp-state` element must contain a child `<script>` element.
 
 ```html
-<!-- Store complex nested JSON data in <amp-state> elements. -->
-<amp-state id="myAnimals">
+<amp-state id="myLocalState">
   <script type="application/json">
     {
-      "dog": {
-        "imageUrl": "/img/dog.jpg",
-        "style": "greenBackground"
-      },
-      "cat": {
-        "imageUrl": "/img/cat.jpg",
-        "style": "redBackground"
+      "foo": "bar"
+    }
+  </script>
+</amp-state>
+```
+
+[/filter] <!-- formats="email" -->
+
+[filter formats="websites, stories, ads"]
+
+#### Attributes
+
+[filter formats="websites, stories, ads"]
+
+##### `src` (optional)
+
+The URL of the remote endpoint that will return the JSON and update this `amp-state`. This must be a CORS HTTP service. The `src` attribute allows all standard URL variable substitutions. See the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info.
+
+[tip type="important"]
+The endpoint must implement the requirements specified in the [CORS Requests in AMP](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests) spec.
+[/tip]
+
+##### `credentials` (optional)
+
+Defines a `credentials` option as specified by the [Fetch API](https://fetch.spec.whatwg.org/).
+
+- Supported values: `omit`, `include`
+- Default: `omit`
+
+To send credentials, pass the value of `include`. If this value is set, the response must follow the [AMP CORS security guidelines](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests/#cors-security-in-amp).
+[/filter] <!-- formats="websites, stories, ads" -->
+
+[filter formats="email"]
+
+#### Actions
+
+##### `refresh`
+
+The `refresh` action refetches data from data point the `src` attribute points to. It ignores the browser cache.
```

Comment:
```suggestion
The `refresh` action refetches data from data point the `src` attribute points to. This action will make a network request bypassing the browser's caching mechanisms.
```

Reply:
Also in the future we might include how this request bypasses the network cache (I assume it's via `Math.random` query parameter).